TelepresenceBot/mobile/src/main/java/com/novoda/tpbot/bot/BotActivity.java

```diff
@@ -35,6 +50,68 @@ protected void onCreate(Bundle savedInstanceState) {
 
         RecyclerView.Adapter adapter = new DirectionAdapter(LayoutInflater.from(this), Collections.<Direction>emptyList());
         directions.setAdapter(adapter);
+
+        ControllerView controllerView = Views.findById(this, R.id.bot_debug_controls);
+        controllerView.setControllerListener(controllerListener);
+
+        handler = new Handler();
+    }
+
+    private final ControllerListener controllerListener = new ControllerListener() {
+
+        @Override
+        public void onDirectionPressed(Direction direction) {
+            startRepeatingCommand(direction.rawCommand());
+        }
+
+        @Override
+        public void onDirectionReleased(Direction direction) {
+            stopRepeatingCommand(direction.rawCommand());
+        }
+
+        @Override
+        public void onLazersFired() {
+            // no-op for debug
+        }
+
+        @Override
+        public void onLazersReleased() {
+            // no-op for debug
+        }
+    };
+
+    private void startRepeatingCommand(String command) {
```

Comment:
I am thinking about encapsulating this command handling in a separate class. 
You would have a `CommandHandler` as a field maybe which has `currentCommand` inside.
I think it is weird to see `String currentCommand` inside the Activity.